CONTRIBUTING.md

```diff
@@ -0,0 +1,152 @@
+
+### Highlight your code snippets in your messages
+
+Wrap your message in double quotes and escape the back ticks.
+
+```
+git commit -m "feat: add \`boolean\` prop \`open\` to \`<dialog>\`"
+```
+
+---
+
+## Release Process
+
+### 1) Feature Branches (regular changes)
+
+The most common process is to branch off from **main**, add your code and merge it to **main** with `Rebase and merge`
+option. [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
+and [Semantic Releases](https://github.com/semantic-release/semantic-release) will take care of the rest. See image
+below for visual clarity:
+
+<img src="https://user-images.githubusercontent.com/2275908/193548546-ddc9582d-d724-4903-a531-71aa3c5bf1ac.png" alt="Release Flow" width="600">
+
+### 2) Pre-Releases (Beta / Release Candidate branches for complex changes)
+
+(See image below for visual clarity)
+
+-   Branch off from `main` to your `feature-branch`.
+-   Commit the code you need and do your local tests.
+-   Open a PR and wait for approval.
+-   Checkout `beta` branch and `cherry-pick` your changes from `feature-branch`:
+    -   If you previously checked out beta, always ensure it is always up-to-date with `main` before `cherry-pick`'ing. By doing it, you release a `beta` version based on the current `main` branch + your cherry-picked changes.
+    -   **NOTE:** Everytime a push in beta occurs a new version gets deployed in NPM. Changelog, Package.json and Lock files also change. Update your local beta branch after the release has finished.
+    -   Use the beta package to test your application.
+    -   **NOTE:** ⛔ Never merge `beta` in `main`.
+-   After tests are complete you either:
+    -   Merge your `feature-branch` in `main`.
+    -   `cherry-pick` your `feature-branch` commits into an `rc` branch to hold / group those for a release in the upcoming weeks.
+
+<img src="https://user-images.githubusercontent.com/2275908/193548740-692a3567-d305-4437-b298-742a2e97980d.png" alt="Pre-Releases Flow" width="600">
+
+### 3) Experimental Releases
+
+(See image below for visual clarity)
+
+Similar to pre-releases (2), they're not required to either be tested through beta or rc branches. They are one off simpler and less complex experimental features.
+
+-   Branch off from `main` to your `feature-branch`.
+-   Commit the code you need and do your local tests.
+-   Open a PR and wait for approval.
+-   Branch off from `feature-branch` to a branch prepended with `exp-*`. eg. `exp-feature-branch`.
+-   When pushing your changes a new experimental release is promoted. eg. `1.3.0-exp-feature-branch.1`.
+-   **NOTE:** ⛔ Never merge `exp-*` branches in `main`.
+-   After tests are complete **delete** the `exp-*` branch in origin.
+
+<img src="https://user-images.githubusercontent.com/2275908/195072598-be6c6ab4-2be1-4f74-941f-5f4ff0337ed0.png" alt="Experimental Releases" width="600">
```

Comment:
🟢 I don't think we should rely on this URL always being available. If we require an image in a README, we should also host that image directly in the same repo.

Reply:
I agree, not the best solution. Probably should reference local file.

For now I'll leave as is but will think about how to best solve this. One way would be to include files in repo and reference locally although that's not really ideal either. Will think about it.